noxfile.py

```diff
@@ -1,153 +1,160 @@
-    serious code quality issues.
+    This currently uses Python 3.6 due to the automated Kokoro run of synthtool.
+    That run uses an image that doesn't have 3.6 installed. Before updating this
+    check the state of the `gcp_ubuntu_config` we use for that Kokoro run.
     """
-    session.install('flake8')
-    session.install('-e', '.')
-    session.run(
-        'flake8',
-        os.path.join('google', 'resumable_media'),
-        'tests',
-    )
+    session.install(BLACK_VERSION)
+    session.run("black", *BLACK_PATHS)
 
 
-@nox.session(python='3.6')
+@nox.session(python="3.7")
 def lint_setup_py(session):
     """Verify that setup.py is valid (including RST check)."""
-    session.install('docutils', 'Pygments')
+    session.install("docutils", "pygments")
+    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")
+
+
+def default(session):
+    # Install all test dependencies, then install this package in-place.
+    session.install("mock", "pytest", "pytest-cov")
+    for local_dep in LOCAL_DEPS:
+        session.install("-e", local_dep)
+    session.install("-e", ".")
+
+    # Run py.test against the unit tests.
     session.run(
-        'python', 'setup.py', 'check', '--restructuredtext', '--strict')
+        "py.test",
+        "--quiet",
+        "--cov=google.cloud",
+        "--cov=tests.unit",
+        "--cov-append",
+        "--cov-config=.coveragerc",
+        "--cov-report=",
+        "--cov-fail-under=0",
+        os.path.join("tests", "unit"),
+        *session.posargs,
+    )
+
+
+@nox.session(python=["2.7", "3.5", "3.6", "3.7"])
+def unit(session):
+    """Run the unit test suite."""
+    default(session)
 
 
-@nox.session(python=['2.7', '3.6'])
-def system_tests(session):
+@nox.session(python=["2.7", "3.7"])
+def system(session):
     """Run the system test suite."""
+    system_test_path = os.path.join("tests", "system.py")
+    system_test_folder_path = os.path.join("tests", "system")
+    # Sanity check: Only run tests if the environment variable is set.
+    if not os.environ.get("GOOGLE_APPLICATION_CREDENTIALS", ""):
+        session.skip("Credentials must be set via environment variable")
 
-    # Sanity check: environment variables are set.
-    missing = []
-    for env_var in SYSTEM_TEST_ENV_VARS:
-        if env_var not in os.environ:
-            missing.append(env_var)
+    system_test_exists = os.path.exists(system_test_path)
+    system_test_folder_exists = os.path.exists(system_test_folder_path)
+    # Sanity check: only run tests if found.
+    if not system_test_exists and not system_test_folder_exists:
+        session.skip("System tests were not found")
 
-    # Only run system tests if the environment variables are set.
-    if missing:
-        all_vars = ', '.join(missing)
-        msg = 'Environment variable(s) unset: {}'.format(all_vars)
-        session.skip(msg)
+    # Use pre-release gRPC for system tests.
+    session.install("--pre", "grpcio")
```

Comment:
- Skipping the systests when the environment variable is not set should remain.

- We have no `grpc` in this library.